airflow/models/skipmixin.py

```diff
@@ -66,9 +67,10 @@ def _set_state_to_skipped(self, dag_run: "DagRun", tasks: "Iterable[BaseOperator
     def skip(
         self,
         dag_run: "DagRun",
-        execution_date: "timezone.DateTime",
-        tasks: "Iterable[BaseOperator]",
-        session: "Session" = None,
+        execution_date: "DateTime",
+        tasks: Sequence["BaseOperator"],
+        *,
+        session: "Session",
```

Comment:
We probably shouldn't make session a kw-only arg, as that _might_ break someone's custom operator :(